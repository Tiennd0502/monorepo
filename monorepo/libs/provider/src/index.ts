export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
