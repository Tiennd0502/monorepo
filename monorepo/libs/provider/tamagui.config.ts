import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
