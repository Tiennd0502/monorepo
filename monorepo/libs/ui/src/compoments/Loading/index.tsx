import { memo } from 'react';
import { Spinner, Stack, ColorTokens } from 'tamagui';

interface LoadingProps {
  size?: 'small' | 'large';
  color?: ColorTokens;
}

const Loading = ({ size = 'large', color = '$primary' }: LoadingProps) => (
  <Stack
    alignSelf="center"
    zIndex="$5"
    height="$full"
    width="$full"
    justifyContent="center"
    position="absolute"
  >
    <Spinner size={size} color={color} />
  </Stack>
);

export default memo(Loading);
