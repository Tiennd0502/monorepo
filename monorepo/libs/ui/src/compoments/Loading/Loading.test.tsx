import {render} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import Loading from '.';

describe('Loading', () => {
  it('should render loading indicator component', () => {
    const container = render(
      <TamaguiProvider>
        <Loading />
      </TamaguiProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
