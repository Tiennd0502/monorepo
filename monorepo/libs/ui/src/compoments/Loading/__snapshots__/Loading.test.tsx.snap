// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render loading indicator component 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "height": "100%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "100%",
      "zIndex": 500,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
      }
    }
  >
    <ActivityIndicator
      color="#242424"
      size="large"
    />
  </View>
</View>
`;
